src/append/rolling_file/rolling.rs

```diff
@@ -261,53 +265,81 @@ impl State {
         let read_dir = fs::read_dir(&self.log_dir)
             .with_context(|| format!("failed to read log dir: {}", self.log_dir.display()))?;
 
+        // Determine the base prefix to match against all log files
+        let base_prefix = if let Some(prefix) = &self.log_filename_prefix {
```

Comment:
But the `log_filename_prefix` filter should not contain the datepart and consider all the log files together?